flake.nix

```diff
@@ -18,23 +18,24 @@
       with pkgs;
       {
         devShells.default = mkShell {
+          nativeBuildInputs = lib.optionals (stdenv.isDarwin) [
+            pkgs.darwin.apple_sdk.frameworks.SystemConfiguration
+          ];
           buildInputs = [
             openssl
             pkg-config
-            libudev-zero
             jq
             (rust-bin.stable.latest.default.override {
               extensions = [ "rust-src" ];
               targets = [ "wasm32-unknown-unknown" ];
             })
-          ];
+          ] ++ lib.optionals (stdenv.isLinux) [libudev-zero];
           shellHook =
           ''
             echo "Using `nix --version`"
             alias stellar="cargo run --bin stellar --"
-            [ -f ./local.sh ] && source ./local.sh
             shell=$0
-            shell=`basename $SHELL`
+            shell=`basename $shell`
             source <(stellar completion --shell $shell)
```

Comment:
I'm also unsure why or if we should have this line. This means that when you use the flake in this directory you automatically start compiling and setting up completions for the project, which seems surprising, and like the completions would get out of date if I'm working on multiple branches and swapping back and forth.

Reply:
We can remove it, I don't have any particular preferences. I mostly was writing it for myself, as there wasn't a demand for it. Now that we see people actually using it, I'm all down to make it user-friendly. So we can remove shell completions as well